DalStatement.php

```diff
@@ -47,6 +47,115 @@
 class DalStatement
 {
     /**
+     * Fetch the next row in the result set.
+     *
+     * @var int
+     */
+    const FETCH_FORWARD      = 0;
+
+    /**
+     * Fetch the previous row in the result set.
+     *
+     * @var int
+     */
+    const FETCH_BACKWARD     = 1;
+
+    /**
+     * Start at the first offset.
+     *
+     * @var int
+     */
+    const FETCH_OFFSET_FIRST = 0;
+
+    /**
+     * Start at the last offset.
+     *
+     * @var int
+     */
+    const FETCH_OFFSET_LAST  = -1;
+
+    /**
+     * Specifies that the fetch method shall return each row as an object with
+     * variable names that correspond to the column names returned in the result
+     * set. FETCH_LAZY creates the object variable names as they are accessed.
+     *
+     * @var int
+     */
+    const FETCH_LAZY  = 1;
+
+    /**
+     * Specifies that the fetch method shall return each row as an array indexed
+     * by column name as returned in the corresponding result set. If the result
+     * set contains multiple columns with the same name, FETCH_ASSOC returns
+     * only a single value per column name.
+     *
+     * @var int
+     */
+    const FETCH_ASSOC = 2;
+
+    /**
+     * Specifies that the fetch method shall return each row as an array indexed
+     * by column number as returned in the corresponding result set, starting at
+     * column 0.
+     *
+     * @var int
+     */
+    const FETCH_NUM = 3;
+
+    /**
+     * Specifies that the fetch method shall return each row as an array indexed
+     * by both column name and number as returned in the corresponding result
+     * set, starting at column 0.
+     *
+     * @var int
+     */
+    const FETCH_BOTH = 4;
+
+    /**
+     * Specifies that the fetch method shall return each row as an object with
+     * property names that correspond to the column names returned in the result
+     * set.
+     *
+     * @var int
+     */
+    const FETCH_OBJ = 5;
+
+    /**
+     * Specifies that the fetch method shall return TRUE and assign the values
+     * of the columns in the result set to the PHP variables to which they were
+     * bound with the DalStatement::bindParameter method.
+     *
+     * @var int
+     */
+    const FETCH_BOUND = 6;
+
+    /**
+     * Specifies that the fetch method shall return only a single requested
+     * column from the next row in the result set.
+     *
+     * @var int
+     */
+    const FETCH_COLUMN = 7;
+
+    /**
+     * Specifies that the fetch method shall return a new instance of the
+     * requested class, mapping the columns to named properties in the class.
+     * The magic __set method is called if the property doesn't exist in the
+     * requested class.
+     *
+     * @var int
+     */
+    const FETCH_CLASS = 8;
```

Comment:
> FROM_CLASS_NAME_IN_FIRST_COLUMN? What is it :smiley:?

I forgot to push my work last time, will do it now :)

I don't like use custom alias, `$stmt::FORWARD` is great thanks.